flang/lib/Lower/DirectivesCommon.h

```diff
@@ -411,12 +417,22 @@ void genOmpAccAtomicRead(Fortran::lower::AbstractConverter &converter,
   mlir::Value toAddress = fir::getBase(converter.genExprAddr(
       *Fortran::semantics::GetExpr(assignmentStmtVariable), stmtCtx));
   fir::FirOpBuilder &builder = converter.getFirOpBuilder();
-  if (fromAddress.getType() != toAddress.getType())
-    fromAddress =
-        builder.create<fir::ConvertOp>(loc, toAddress.getType(), fromAddress);
-  genOmpAccAtomicCaptureStatement(converter, fromAddress, toAddress,
+  mlir::Value LHSTemp = toAddress;
+  if (fromAddress.getType() != toAddress.getType()) {
+    // Create a temp for implicit type conversion.
+    LHSTemp = builder.create<fir::AllocaOp>(
```

Comment:
You can use `fir::FirOpBuilder::getAllocaBlock` to find an appropriate place to insert an alloca.

Putting it here runs the risk of allocating stack space inside of a loop.

Reply:
Thank you for your comment. I tried `createTemporary` in the FIR builder. But that caused an issue in the operation walk.  Maybe I could take a different approach. This line was already deleted in the latest change.